src/vs/code/electron-main/windows.ts

```diff
@@ -729,9 +744,16 @@ export class WindowsManager implements IWindowsMainService {
 				state: this.getNewWindowState(configuration),
 				extensionDevelopmentPath: configuration.extensionDevelopmentPath,
 				allowFullscreen: this.lifecycleService.wasUpdated || (windowConfig && windowConfig.restoreFullscreen),
-				titleBarStyle: windowConfig ? windowConfig.titleBarStyle : void 0
+				titleBarStyle: windowConfig ? windowConfig.titleBarStyle : void 0,
+				vscodeWindowId: configuration.vscodeWindowId
 			});
 
+			configuration.vscodeWindowId = vscodeWindow.vscodeWindowId;
+			if (!configuration.extensionDevelopmentPath) {
+				// TODO: Cover closing a folder/existing window case
+				this.backupService.pushEmptyWorkspaceBackupWindowIdSync(configuration.vscodeWindowId);
```

Comment:
@Tyriar I suggest to move this up to the place where we already call `pushWorkspaceBackupPathsSync` so that we have it in one place. doing this from `toConfiguration` is very weird. As a caller of this method I would expect to get a configuration object but as a side effect I am writing to the backup folder??